.gitlab-ci.yml

```diff
@@ -132,7 +132,7 @@ common-stg-test-gcp-dev:
   after_script:
     # Clean up dev environment even if one of the previous steps fails
     - cd gcp/live/dev
-    - rake destroy || true
+    - rake destroy
```

Comment:
This change means that if `rake destroy` fails, then none of the subsequent commands will run. Skipping `rake destroy_say_keys` and `rake clobber` doesn't look right to me.

I proposed a different solution in https://issues.gpii.net/browse/GPII-3488.

Reply:
I still think that the best course of action when `rake destroy` fails is to do nothing and leave things as they are, so we can investigate what happened. 

Agree that keeping TF state is crucial for this, but secrets data can also be pretty important. I guess we can still run `rake destroy_sa_keys` and `rake clobber` safely...

Does it look good to you if I replace `rake destroy || true` after script with `(rake destroy && rake destroy_tfstate[k8s] && rake destroy_secrets) || true`, so subsequent commands `rake destroy_tfstate[locust]`, `rake destroy_sa_keys`, `rake clobber` can still run?

Reply:
> I still think that the best course of action when rake destroy fails is to do nothing and leave things as they are, so we can investigate what happened.

I think there is merit in this argument. The main counterargument I see is the cost of leaving a dev cluster running for N additional hours until someone investigates a failure or a new pipeline runs. It is possible that this cost is negligible and outweighed by the value of being able to investigate the cluster that produced the failed build. This topic may warrant further discussion.

If we choose to change this behavior we should add a `rake destroy` before a dev Job's initial `rake` so that it is deploying into a (hopefully) clean environment.

> replace rake destroy || true after script with (rake destroy && rake destroy_tfstate[k8s] && rake destroy_secrets) || true, so subsequent commands rake destroy_tfstate[locust], rake destroy_sa_keys, rake clobber can still run?

Yes, this is what I was proposing.

Reply:
> The main counterargument I see is the cost of leaving a dev cluster running for N additional hours

If destroy task fails on some dependency of `k8s-cluster` module, cluster is left running and consuming resources anyway. 

> If we choose to change this behavior we should add a rake destroy before a dev Job's initial rake so that it is deploying into a (hopefully) clean environment.

I was thinking about that, but IMO, if `rake destroy` fails, chances that the failure is intermittent and going to be solved by another destroy without manual intervention are pretty low.